src/controllers/update/overseas.entity.query.controller.ts

```diff
@@ -29,6 +30,12 @@ export const post = async (req: Request, res: Response, next: NextFunction) => {
   try {
     logger.debugRequest(req, `${req.method} ${req.route.path}`);
 
+    const appData: ApplicationData = getApplicationData(req.session);
+    // For Resume cases appData will contain overseas entity id so don't call getCompanyProfile
+    if (appData.overseas_entity_id) {
```

Comment:
It's the entity_number that contains OExxxxxx. That's when you don't call getCompanyProfile. The overseas_entity_id is set on first save?

Reply:
@AndreKramer7654 , was only considering Resume cases and for Resume cases the overseas_entity_id will be present and the first save happens after confirming the company details

Reply:
Ah, code below. You are skipping this page and only showing confirm. That needs product input. I was assuming pre-populated query dialog that can't be changed (on resume) - not skipping altogether. Personally, I see no reason not to be able to change the OE number after resume but again, product decision, I think.

Reply:
There is an option of going back and choosing another company hence did not add the entity_number check , the logic for resetting everything is already present

Reply:
> With this the user will never be able to change the OE number after they confirm - not just for resume. Is that really what we want?

if its before the transaction and overseas-entity-id is created, they will have the option of going back and changing it , if after then yes they will not be able to change , previously it was suggested that the field should be made not editable but that decision is yet pending

Reply:
Let's remove this if (appData.overseas_entity_id) for now.